provider/datagen/src/bin/datagen.rs

```diff
@@ -2,13 +2,194 @@
+    verbose: bool,
+
+    #[arg(long, value_enum, default_value_t = Format::DeprecatedDefault, hide_default_value = true)]
+    #[arg(
+        help = "Select the output format: a directory tree of files, a single blob, or a Rust module."
+    )]
+    format: Format,
+
+    #[arg(short = 'W', long)]
+    #[arg(help = "Delete the output before writing data.")]
+    overwrite: bool,
+
+    #[arg(short, long, value_enum)]
+    #[arg(help = "--format=dir only: serde serialization format.")]
+    syntax: Syntax,
+
+    #[arg(short, long)]
+    #[arg(help = "--format=mod, --format=dir only: pretty-print the Rust or JSON output files.")]
+    pretty: bool,
+
+    #[arg(long)]
+    #[arg(help = "--format=dir only: whether to add a fingerprints file to the output.")]
+    fingerprint: bool,
+
+    #[arg(short = 't', long, value_name = "TAG", default_value = "latest")]
+    #[arg(
+        help = "Download CLDR JSON data from this GitHub tag (https://github.com/unicode-org/cldr-json/tags)\n\
+                    Use 'latest' for the latest version verified to work with this version of the binary.\n\
+                    Ignored if '--cldr-root' is present.\n\
+                    Note that some keys do not support versions before 41.0.0."
+    )]
+    cldr_tag: String,
+
+    #[arg(long, value_name = "PATH")]
+    #[arg(
+        help = "Path to a local cldr-{version}-json-full.zip directory (see https://github.com/unicode-org/cldr-json/releases).\n\
+                  Note that some keys do not support versions before 41.0.0."
+    )]
+    cldr_root: PathBuf,
+
+    #[arg(long, value_name = "TAG", default_value = "latest")]
+    #[arg(
+        help = "Download Unicode Properties data from this GitHub tag (https://github.com/unicode-org/icu/tags)\n\
+                  Use 'latest' for the latest version verified to work with this version of the binary.\n\
+                  Ignored if '--icuexport-root' is present.\n\
+                  Note that some keys do not support versions before release-71-1."
+    )]
+    icuexport_tag: String,
+
+    #[arg(long, value_name = "PATH")]
+    #[arg(
+        help = "Path to a local icuexportdata_uprops_full directory (see https://github.com/unicode-org/icu/releases).\n\
+                  Note that some keys do not support versions before release-71-1."
+    )]
+    icuexport_root: PathBuf,
+
+    #[arg(long, value_enum, default_value_t = TrieType::Small)]
+    #[arg(
+        help = "Whether to optimize CodePointTrie data structures for size (\"small\") or speed (\"fast\").\n\
+                  Using \"fast\" mode increases performance of CJK text processing and segmentation. For more\n\
+                  information, see the TrieType enum."
+    )]
+    trie_type: TrieType,
+
+    #[arg(long, value_enum, default_value_t = CliCollationHanDatabase::Implicit)]
+    #[arg(help = "Which collation han database to use.")]
+    collation_han_database: CliCollationHanDatabase,
```

Comment:
Thought: I'm not super excited about tools (here or elsewhere) that change the casing of strings. It is more clear if the literal strings are in the code.

Reply:
I kinda like the conciseness of relying on the magic 99% of the time. It makes the deviations clearer